README
NAME
    Dist::Zilla::Plugin::NextVersion::Semantic - update the next version,
    semantic-wise

VERSION
    version 0.2.4

SYNOPSIS
        # in dist.ini

        [NextVersion::Semantic]
        major = MAJOR, API CHANGE
        minor = MINOR, ENHANCEMENTS
        revision = REVISION, BUG FIXES

        ; must also load a PreviousVersionProvider
        [PreviousVersion::Changelog]

DESCRIPTION
    Increases the distribution's version according to the semantic
    versioning rules (see <http://semver.org/>) by inspecting the changelog.

    More specifically, the plugin performs the following actions:

    at build time
        Reads the changelog using "CPAN::Changes" and filters out of the
        "{{$NEXT}}" release section any group without item.

    before a release
        Ensures that there is at least one recorded change in the changelog,
        and increments the version number in consequence. If there are
        changes given outside of the sections, they are considered to be
        minor.

    after a release
        Updates the new "{{$NEXT}}" section of the changelog with
        placeholders for all the change categories. With categories as given
        in the *SYNOPSIS*, this would look like

            {{$NEXT}}

              [MAJOR]

              [API CHANGE]

              [MINOR]

              [ENHANCEMENTS]

              [REVISION]

              [BUG FIXES]

    If a version is given via the environment variable "V", it will taken
    as-if as the next version.

    For this plugin to work, your Dist::Zilla configuration must also
    contain a plugin consuming the
    Dist::Zilla::Role::YANICK::PreviousVersionProvider role.

    In the different configuration attributes where change group names are
    given, the special group name "UNGROUPED" can be given to specify the
    nameless group.

        0.1.3 2013-07-18

        - this item will be part of UNGROUPED.

        [BUG FIXES]
        - this one won't.

PARAMETERS
  change_file
    File name of the changelog. Defaults to "Changes".

  numify_version
    If true, the version will be a number using the *x.yyyzzz* convention
    instead of *x.y.z*. Defaults to false.

  format
    Specifies the version format to use. Follows the '%d' convention of
    "sprintf" (see examples below), excepts for one detail: '%3d' won't pad
    with whitespaces, but will only determine the maximal size of the
    number. If a version component exceeds its given size, the next version
    level will be incremented.

    Examples:

        %d.%3d.%3d 
            PATCH LEVEL INCREASES: 0.0.998 -> 0.0.999 -> 0.1.0
            MINOR LEVEL INCREASES: 0.0.8 -> 0.1.0 -> 0.2.0
            MAJOR LEVEL INCREASES: 0.1.8 -> 1.0.0 -> 2.0.0

        %d.%02d%02d
            PATCH LEVEL INCREASES: 0.0098 -> 0.00099 -> 0.0100
            MINOR LEVEL INCREASES: 0.0008 -> 0.0100 -> 0.0200
            MAJOR LEVEL INCREASES: 0.0108 -> 1.0000 -> 2.0000

        %d.%05d
            MINOR LEVEL INCREASES: 0.99998 -> 0.99999 -> 1.00000
            MAJOR LEVEL INCREASES: 0.00108 -> 1.00000 -> 2.00000

    Defaults to '%d.%3d.%3d'.

  major
    Comma-delimited list of categories of changes considered major. Defaults
    to "API CHANGES".

  minor
    Comma-delimited list of categories of changes considered minor. Defaults
    to "ENHANCEMENTS" and "UNGROUPED".

  revision
    Comma-delimited list of categories of changes considered revisions.
    Defaults to "BUG FIXES, DOCUMENTATION".

AUTHOR
    Yanick Champoux <yanick@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2012 by Yanick Champoux.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for {{$dist->name}}

{{$NEXT}}
 [API CHANGES]

 [BUG FIXES]

 [DOCUMENTATION]

 [ENHANCEMENTS]

 [NEW FEATURES]

 [STATISTICS]

0.2.6 2021-01-18
 [BUG FIXES]
 - Don't let tests fail when V=1 is in MAKEOPTS for Gnu AutoTools "Verbose"
   mode.
   [#6](https://github.com/yanick/Dist-Zilla-Plugin-NextVersion-Semantic/pull/6),
   thanks to kentfredric.

 [STATISTICS]
 - code churn: 7 files changed, 202 insertions(+), 44 deletions(-)

0.2.5 2015-06-15
 [BUG FIXES]
 - Silence Perl 5.22 warnings. (GH#5)

 [STATISTICS]
 - code churn: 5 files changed, 22 insertions(+), 9 deletions(-)

0.2.4 2015-05-24
 [BUG FIXES]
 - Mix of patch and minor was giving wrong increment. (GH#4)

 [STATISTICS]
 - code churn: 4 files changed, 30 insertions(+), 5 deletions(-)

0.2.3 2015-01-22
 [BUG FIXES]
 - Only support Perl 5.10.0 and up.
 - distro was claiming DZT::Sample.

 [STATISTICS]
 - code churn: 3 files changed, 14 insertions(+), 2 deletions(-)

0.2.2 2014-12-08
 [BUG FIXES]
 - Custom version formats didn't work on  first version. (GH#3, Csson)

 [STATISTICS]
 - code churn: 3 files changed, 21 insertions(+), 17 deletions(-)

0.2.1 2014-11-08
 [BUG FIXES]
 - Fix error message.

 [STATISTICS]
 - code churn: 3 files changed, 12 insertions(+), 3 deletions(-)

0.2.0 2014-08-31
 [ENHANCEMENTS]
 - Version format is now configurable.

 [STATISTICS]
 - code churn: 1 file changed, 7 insertions(+), 3 deletions(-)

0.1.3 2013-07-18
 [BUG FIXES]
 - remove use of smartmatches to silence 5.18's warnings

 [STATISTICS]
 - code churn: 4 files changed, 95 insertions(+), 86 deletions(-)

0.1.2 2013-05-02
 [BUG FIXES]
 - Bump CPAN::Changes dependency to v0.20.

 [STATISTICS]
 - code churn: 2 files changed, 23 insertions(+), 6 deletions(-)

0.1.1 2013-05-01
 [BUGS FIXES]
 - Forgot BeforeRelease role (oops). (RT#84744, patch by Alex Peters)
 - Clean up error messages and improve docs for PreviousVersionProvider.
   (GH#1, patch by Alex Peters)

 [STATISTICS]
 - code churn: 6 files changed, 36 insertions(+), 12 deletions(-)

0.1.0 2012-03-19
 - release the Kraken!


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.031.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.28"
  },
  "configure_requires" => {
    "Module::Build" => "0.28"
  },
  "dist_abstract" => "update the next version, semantic-wise",
  "dist_author" => [
    "Yanick Champoux <yanick\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-NextVersion-Semantic",
  "dist_version" => "0.2.3",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::NextVersion::Semantic",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "CPAN::Changes" => "0.20",
    "Dist::Zilla::Role::AfterRelease" => 0,
    "Dist::Zilla::Role::BeforeRelease" => 0,
    "Dist::Zilla::Role::FileMunger" => 0,
    "Dist::Zilla::Role::Plugin" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Dist::Zilla::Role::VersionProvider" => 0,
    "List::AllUtils" => 0,
    "List::Util" => 0,
    "Moose" => 0,
    "Moose::Role" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "Perl::Version" => 0,
    "perl" => "v5.10.0",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::DZil" => 0,
    "Test::Exception" => 0,
    "Test::More" => "0.88"
  }
);


my %fallback_build_requires = (
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Module::Build" => "0.28",
  "Test::DZil" => 0,
  "Test::Exception" => 0,
  "Test::More" => "0.88"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.034.
use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "update the next version, semantic-wise",
  "AUTHOR" => "Yanick Champoux <yanick\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-NextVersion-Semantic",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010000",
  "NAME" => "Dist::Zilla::Plugin::NextVersion::Semantic",
  "PREREQ_PM" => {
    "CPAN::Changes" => "0.20",
    "Dist::Zilla::Role::AfterRelease" => 0,
    "Dist::Zilla::Role::BeforeRelease" => 0,
    "Dist::Zilla::Role::FileMunger" => 0,
    "Dist::Zilla::Role::Plugin" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Dist::Zilla::Role::VersionProvider" => 0,
    "List::AllUtils" => 0,
    "List::Util" => 0,
    "Moose" => 0,
    "Moose::Role" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "Perl::Version" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::DZil" => 0,
    "Test::Exception" => 0,
    "Test::More" => "0.88"
  },
  "VERSION" => "0.2.4",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "CPAN::Changes" => "0.20",
  "Dist::Zilla::Role::AfterRelease" => 0,
  "Dist::Zilla::Role::BeforeRelease" => 0,
  "Dist::Zilla::Role::FileMunger" => 0,
  "Dist::Zilla::Role::Plugin" => 0,
  "Dist::Zilla::Role::TextTemplate" => 0,
  "Dist::Zilla::Role::VersionProvider" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "List::AllUtils" => 0,
  "List::Util" => 0,
  "Moose" => 0,
  "Moose::Role" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "Perl::Version" => 0,
  "Test::DZil" => 0,
  "Test::Exception" => 0,
  "Test::More" => "0.88",
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
